core/trino-main/src/main/java/io/trino/simd/BlockEncodingSimdSupport.java

```diff
@@ -32,6 +31,13 @@ We need to specifically detect AVX512F (for VPCOMPRESSD / VPCOMPRESSQ instructio
 whether Vector<T>#compress(VectorMask<T>) is supported natively in hardware or emulated by the JVM - because the emulated
 support is so much slower than the simple scalar code that exists, but since we don't have the ability to detect that
 directly from the JDK vector API we have to assume that native support exists whenever the CPU advertises it.
+
+On ARM, the situation is similar but the threshold for useful performance is different: NEON does not provide a native
+equivalent for the vector compress operation and JVM emulation tends to be significantly slower than our scalar fallback.
+SVE, on the other hand, offers predicate-driven operations that can lower Vector<T>#compress to efficient native
+instructions. Since the JDK vector API does not expose whether compress is lowered to SVE or emulated, we conservatively
+enable SIMD-only paths on ARM when the CPU advertises SVE (and the preferred vector width is large enough), and treat
+NEON-only systems as not supporting native compress for the purposes of this optimization.
```

Comment:
It would be good to mention which graviton generations are NEON only and which have SVE.